tensorflow_datasets/core/custom_dataset/image_folder.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Image FileFolder datasets."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import itertools
+import os
+
+from absl import logging
+import tensorflow.compat.v2 as tf
+from tensorflow_datasets import core
+
+_SUPPORTED_IMAGE_FORMAT = ('.jpg', '.jpeg', '.png')
+_AUTOTUNE = tf.data.experimental.AUTOTUNE
+
+
+class ImageLabelFolder(core.DatasetBuilder):
+
+  VERSION = core.Version(
+    "2.0.0", "New split API (https://tensorflow.org/datasets/splits)")
+
+  def __init__(self, data_dir):
+    self._data_dir = os.path.expanduser(data_dir)
+
+    # dict[split_name][label_name] = list(img_paths)
+    self._split_label_images, self._labels = self._get_split_label_images()
+    self._splits_info = core.SplitDict(self.name)
+
+    # Calculate splits for DatasetInfo
+    self._splits = {
+        split_name: self._get_split_size(split_name)
+        for split_name in self._split_label_images
+    }
+
+    self._split_dict = core.SplitDict(self.name)
```

Comment:
We should try avoid adding too many new `self._` attributes. Some seems to duplicate information (e.g. `self._splits_dict` VS `self.info.splits` VS `self._splits`).
I think they could be merged to only use `self.info.splits`.

Reply:
Removed the `self._` and made them local variables.
I think they could be merged to only use `self.info.splits` because they are class properties.